api/v1/intrusiondetection_types.go

```diff
@@ -27,8 +27,40 @@ type IntrusionDetectionSpec struct {
 	// Only DeepPacketInspection is supported for this spec.
 	// +optional
 	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
+
+	// AnomalyDetectionSpec provides configuration for running AnomalyDetection Component within
+	// IntrusionDetection
+	AnomalyDetectionSpec *AnomalyDetectionSpec `json:"anomalyDetectionSpec,omitempty"`
+}
+
+type AnomalyDetectionSpec struct {
+	// StorageType Sets the type of storage to use for storing Anomaly Detection Models. By default it will use the ephemeral
+	// emptyDir on the node AD will be deployed to. This field sets precedent for all AD Storage related fields and will only
```

Comment:
> Anomaly Detection Models
models (lowercase)

> This field sets precedent for all AD Storage related fields 
I think we better refrain from saying this, since other Operaror overrides are making its way into the operator.

Reply:
I think you still need  `+kubebuilder:validation:Enum` to ensure that the CRD validates the valid enum options